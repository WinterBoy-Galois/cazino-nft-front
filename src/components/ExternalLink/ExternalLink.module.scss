@import '../../styles/shared';

.icon {
  width: 12px;
  height: 12px;
  margin-left: 0.4rem;
}

.link {
  color: $white;
  display: flex;
  align-items: center;

  &:hover {
    color: $white;
  }
}
