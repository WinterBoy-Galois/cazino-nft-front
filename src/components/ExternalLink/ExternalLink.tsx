import React from 'react';
import clsx from 'clsx';
import styles from './ExternalLink.module.scss';
import External from '../icons/External';

interface IProps {
  href: string;
  className?: string;
}

const ExternalLink: React.FC<IProps> = ({ className, href, children }) => {
  return (
    <a
      className={clsx(styles.link, className, 'truncate')}
      href={href}
      target="_blank"
      rel="noopener noreferrer"
    >
      <span className="truncate">{children}</span>
      <External className={styles.icon} />
    </a>
  );
};

export default ExternalLink;
