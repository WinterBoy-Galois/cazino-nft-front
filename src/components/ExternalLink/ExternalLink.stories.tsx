import React from 'react';
import { text } from '@storybook/addon-knobs';

import ExternalLink from '.';

export default {
  title: 'Components/ExternalLink',
  component: ExternalLink,
};

const data = {
  href: 'https://test.de',
};

export const Default = () => <ExternalLink href={text('Href', data.href)} />;
