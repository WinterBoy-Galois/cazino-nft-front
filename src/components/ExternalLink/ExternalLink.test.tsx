import React from 'react';
import { render } from '@testing-library/react';

import ExternalLink from '.';

describe('ExternalLink', () => {
  it('should match snapshot', () => {
    // Arrange

    // Act
    const container = render(<ExternalLink href="https://test.de" />);

    // Assert
    expect(container).toMatchSnapshot();
  });
});
