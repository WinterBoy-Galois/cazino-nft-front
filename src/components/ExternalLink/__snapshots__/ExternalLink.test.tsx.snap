// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalLink should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="link truncate"
        href="https://test.de"
        rel="noopener noreferrer"
        target="_blank"
      >
        <span
          class="truncate"
        />
        <svg
          class="icon"
          viewBox="0 0 12 12"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            data-name="Layer 2"
          >
            <g
              data-name="Layer 1"
            >
              <path
                class="fill"
                d="M0 0v12h12V9h-1.5v1.5h-9v-9H3V0zm6 0l2.25 2.25L4.5 6 6 7.5l3.75-3.75L12 6V0z"
                data-name="Icon open-external-link"
              />
            </g>
          </g>
        </svg>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="link truncate"
      href="https://test.de"
      rel="noopener noreferrer"
      target="_blank"
    >
      <span
        class="truncate"
      />
      <svg
        class="icon"
        viewBox="0 0 12 12"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          data-name="Layer 2"
        >
          <g
            data-name="Layer 1"
          >
            <path
              class="fill"
              d="M0 0v12h12V9h-1.5v1.5h-9v-9H3V0zm6 0l2.25 2.25L4.5 6 6 7.5l3.75-3.75L12 6V0z"
              data-name="Icon open-external-link"
            />
          </g>
        </g>
      </svg>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
